docs/doc/30-reference/30-sql/00-ddl/20-table/60-optimize-table.md

```diff
@@ -2,41 +2,97 @@
+The optimization merges both segments and blocks of a table by default. However, you can choose to compact the segments only when a table has too many segments. 
 
-  If data keeps being injected into table at small scale, and historical data is not required, it is recommended to execute this statement periodically.
+Databend recommends optimizing the segments only for a table when the table has more than 1,000 segments and the average number of blocks per segment is less than 500.
 
+```sql
+select count(*),avg(block_count),if(avg(block_count)<500,'The table needs segment compact now','The table does not need segment compact now') from fuse_segment('<your_database>','<your_table>');
+```
+
+### When NOT to Optimize
+
+Optimizing a table could be time-consuming, especially for large ones. Databend does not recommend optimizing a table too frequently. Before you optimize a table, make sure the table fully satisfies the conditions described in [When to Optimize](#when-to-optimize).
+
+Databend does not compact a table if the number of its blocks is less than two times the [max_threads](../../../30-sql/70-system-tables/system-settings.md#max_threads) value. Databend will not do anything for such tables even though you run the optimization commands on them.
+
+## Syntax
+
+```sql
+OPTIMIZE TABLE [database.]table_name [ PURGE | COMPACT | ALL ] [SEGMENT] [LIMIT <segment_count>]
+```
+
+- `OPTIMIZE TABLE <table_name> PURGE`
+
+    Purges the historical data of table. Only the latest snapshot (including the segments and blocks referenced by this snapshot) will be kept.
  
-- `OPTIMIZE TABLE T COMPACT`
+- `OPTIMIZE TABLE <table_name> COMPACT [LIMIT <segment_count>]`
  
-  Compact the table data only by merging small blocks/segments into larger ones.
+    Compacts the table data by merging small blocks and segments into larger ones.
  
-  - A new snapshot of table T will be added to the history, by this compaction operation.
+    - A new snapshot of table will be created and added to the history, by this compaction operation.
+    - Depending on the size of the given table, it may take quite a while to complete the execution.
+    - The option LIMIT sets the maximum number of segments to be compacted. In this case, Databend will select them segments randomly.
 
-  - Depends on the size of the given table, it may take quite a while to complete the execution.
+-  `OPTIMIZE TABLE <table_name> COMPACT SEGMENT [LIMIT <segment_count>]`
 
- 
-- `optimize table T ALL`
- 
-  Compact the historical data, and then, purge the history 
+    Compacts the table data by merging small segments into larger ones.
+
+    - See [Compacting Segments Only](#compacting-segments-only) for when you need this command.
+    - The option LIMIT sets the maximum number of segments to be compacted. In this case, Databend will select the segments randomly.
```

Comment:
Segments selected is not random, it always selected latest segments

Reply:
It says "random" here:https://github.com/datafuselabs/databend/issues/8241